### DISCUSSION TITLE
Chatgpt會取代工程師的工作嗎?

### DISCUSSION THREAD

**Discussion by HsuSungTing** [1101-GS4719-C/Python-Community-GS4719-C#1153] [General]
我是電機系的徐松廷，今天想和大家分享關於AI是否會取代工程師工作的議題:
        ChatGPT是OpenAI開發的一種自然語言處理模型，可以進行自然語言對話和生成各種文本。ChatGPT可以模仿人類的對話方式，並且在各種自然語言處理任務中取得了極高的表現。由於ChatGPT的強大功能，許多人擔心它是否會取代工程師的工作。
       首先，ChatGPT雖然能夠進行自然語言處理，但它僅僅是一種模型，需要有經驗豐富的工程師進行訓練、調參、優化等工作，才能發揮出最佳的性能。此外，ChatGPT目前還存在著許多限制和挑戰，例如在理解多義詞、處理語言的情感和文化背景等方面還存在著一定的困難，需要有經驗豐富的工程師進行改進和優化。
        其次，ChatGPT僅能應用於自然語言處理領域，並不能替代其他領域的工程師工作，如軟體開發、硬體設計等。即使在自然語言處理領域，ChatGPT也只是一個輔助工具，需要和其他技術、算法結合起來，才能夠實現複雜的應用。

以我的觀點而言，我認為無論是寫code或著是設計硬體，最重要的是如何去定義一個模糊的問題，以及找到在現實中比較好的解決方案，例如開發一個軟體要用什麼技術?要佈署到哪裡?以硬體設計而言，更要了解每個設計的限制進而進行trage off，所以工程師還是很有存在必要，但只是在未來，工程師會更需要有整合性甚至跨域的能力，去作出技術性的決策，因為在一個精確定義的問題中，AI會做得越來越好。

在可預見的未來，AI會大幅度地去輔助工程師的工作，而工程師最終的價值在溝通與協調，進而做出正確的技術性決策去定義每個需要被解決的問題，而已經被完全定義好的問題，可以由AI代勞，所以工程師的需求也可能下降。
如果未來有志成為工程師，我認為學習使用chatgpt非常重要，可以大幅度加速學習，他絕對會是很好的幫手
，至少他教會我Javascript怎麼寫XD。

**Comment by xxrjun** [comment]
蠻認同的，特別是這段「**最重要的是如何去定義一個模糊的問題**」，如何找出想要研究或解決的問題以及透過了解市場需求而開發出被許多人使用的產品，這是GPT技術或是其他大型語言模型目前都做不到的，畢竟GPT技術主要的工作還是文字接龍。不過我自己倒是蠻期待有一天ChatGPT等應用能夠「像是」真的做到前面說的問題，而且讓教授、研究員等學術人員都認(騙)可(過)，或許會有新一代的圖靈測試也說不定(?

勞務類型的工程師(碼農)需求高機率會下降甚至不見，不過市場對於工程師的需求我覺得反而會因為新技術的注入而擴大需求，只是要具備的能力會有不小的轉變，**能否做出決策跟定義問題**將會成為是否會被取代的關鍵。

總體來說，ChatGPT還是非常好用的工具(每月20美已課:money_with_wings:)，結合GitHub Copilot大幅提升學習與開發效率，但盲目地使用工具我認為長期而言是在傷害自己的競爭力，所以在善用工具的同時也要注重同學貼文中提到的點！

**Comment by Lipunpun** [comment]
我是土木系的學生，我非常認同這篇文章的觀點，隨著人工智慧的發展，有人擔心AI會取代工程師的工作。其中，有些人認為自然語言處理模型ChatGPT的出現可能會讓工程師的需求下降，因為AI能夠輔助工程師處理許多問題。然而，這種看法是否正確呢？以下是我自己的一些想法：

第一，雖然ChatGPT可以進行自然語言處理，但它僅僅是一種模型，需要有經驗豐富的工程師進行訓練、調參、優化等工作，才能發揮出最佳的性能。一個好的AI模型背後需要有高質量的數據集和算法設計，以及專業的機器學習知識和經驗。這些都需要工程師的參與，因此ChatGPT不能取代工程師的工作，反而需要他們的協助和指導。

其次，ChatGPT僅能應用於自然語言處理領域，並不能替代其他領域的工程師工作，如軟體開發、硬體設計等。工程師的工作領域很廣泛，AI只是其中的一個輔助工具。即使在自然語言處理領域，ChatGPT也只是一個部分解決方案，需要和其他技術、算法結合起來，才能夠實現複雜的應用。因此，工程師的需求仍然很高，AI不能完全取代他們的工作。

第三，AI在現階段還存在著許多限制和挑戰，例如在理解多義詞、處理語言的情感和文化背景等方面還存在著一定的困難，需要有經驗豐富的工程師進行改進和優化。AI的發展還需要長時間的研究和實驗，這是工程師所擁有的獨特專業知識和技能。

另外也有很多議題，像是倫理道德、用AI跑出來的結果出錯責任歸咎(總不能把AI抓去關吧XD)等等，但用ChatGPT寫文章的架構是真的香(別我怎麼知道的嘿嘿)，總之ChatGPT目前來說真的是一個很好用的輔助工具。

**Comment by shuxian12** [comment]
我覺得目前不能取代的最大問題就在於ChatGPT目前只是依照預設的模型參數產生文字，也可以說是機率的組合，缺乏真正的理解能力，而真正的理解能力對於模型是一個相當困難的問題。例如，在自然語言處理領域中，AI可能無法真正理解人類語言的情感和文化背景，導致產生一些不合理的結果，也導致為何opanAI需要對這些模型進行攻擊性字眼的處理。因此，我認為在近幾年工程師依舊不會被取代，畢竟還是需要人的腦袋，才產生的出來新的模型。

**Reply by HsuSungTing** [reply]
同意，不過我覺得最近真的是變動很大的時期
有些既有的規則會被改變
我也還在摸索在這樣的時代
學習什麼技能是最重要的。
